src/market/scan/scanned-offer.ts

```diff
@@ -145,14 +145,15 @@ export class ScannedOffer {
    * ```
    */
   get paymentPlatformAddresses(): PaymentPlatformAddressSet {
-    const platformProps = Object.entries(this.model.properties).filter(([key]) =>
-      key.startsWith("golem.com.payment.platform."),
-    );
-
-    const platformAddress = platformProps.map(([key, address]) => [
-      key.replace("golem.com.payment.platform.", "").replace(".address", ""),
-      address,
-    ]);
+    const propRegex = /golem.com.payment.platform.([a-z0-9-]+).address/;
```

Comment:
1. `.` should be escaped, right now it matches any character (try `golem☠com☠payment☠platform☠nonsense☠address`)
2. are we sure all constraints keys are lowercase only?

Reply:
AD 1. Will fix that.
AD 2. Yes, that's the case.